shared-bindings/wifi/Radio.c

```diff
@@ -153,6 +170,90 @@ const mp_obj_property_t wifi_radio_hostname_obj = {
               (mp_obj_t)&mp_const_none_obj},
 };
 
+//|     def start_station(self) -> None:
+//|         """Starts a Station."""
+//|         ...
+//|
+STATIC mp_obj_t wifi_radio_start_station(mp_obj_t self) {
+    common_hal_wifi_radio_start_station(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_start_station_obj, wifi_radio_start_station);
+
+//|     def stop_station(self) -> None:
+//|         """Stops the Station."""
+//|         ...
+//|
+STATIC mp_obj_t wifi_radio_stop_station(mp_obj_t self) {
+    common_hal_wifi_radio_stop_station(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_stop_station_obj, wifi_radio_stop_station);
+
+//|     OPEN: int
+//|     WPA_PSK: int
+//|     WPA2_PSK: int
+//|     WPA_WPA2_PSK: int
```

Comment:
You may want to split these out like Pull or Direction in `digitalio`. @jepler did add a way to do enums internally to. That's probably even better. We'll want comments for each entry describing what they mean too.

Reply:
Will definitely add the RTD comments, that was an oversight. Will work on the structure of these tomorrow with jepler's help. Did a little reading, sounds like maybe a scoping issue.

Reply:
> You may want to split these out like Pull or Direction in digitalio.

I did this while working on WiFi Mesh. The changes are in my [`mesh`](https://github.com/microDev1/circuitpython/tree/mesh/shared-bindings/wifi) branch. 
I can create a new PR adding `AuthMode` and then you can do a merge from upstream.

Reply:
Thanks, I'll take a look. I can work from examples like that I think. I'm waiting first to understand what jepler's method is.

Reply:
@microDev1 If you're still game, go for it. I'll need some pointers on how to "merge from upstream". I spent most of the day trying to get one of the methods to work (starting with the hardest), I just don't know these abstractions at all at this point.

Reply:
Sure, I'll make a PR shortly. I think commits after [`c472245`](https://github.com/adafruit/circuitpython/pull/4650/commits/c47224548167d4ab78d884497b68c13bf0560d53) can be cleaned-up?

Reply:
Thank you. Yes, should be fixed now.

Only four authmodes work for AP in the IDF: OPEN, WPA, WPA2, WPA_WPA2.

Reply:
By clean-up. I meant removing those commits.

Reply:
I did the equivalent, re-committed with the code removed.

Reply:
I created PR #4685 adding `AuthMode` class. 
Once it is merged, I'll update this PR enabling setting of authmode for ap.